src/graphql/CompanyUser/Mutations/deleteCompanyUser.ts

```diff
@@ -0,0 +1,33 @@
+import { nonNull, ID, Int } from "$graphql/fieldTypes";
+import { IApolloServerContext } from "$graphql/Context";
+import { UserRepository } from "$models/User";
+import { CompanyUserRepository, DeleteLastCompanyUserError } from "$models/CompanyUser";
+import { UnauthorizedError } from "$graphql/Errors";
+
+export const deleteCompanyUser = {
+  type: Int,
+  args: {
+    uuid: {
+      type: nonNull(ID)
+    }
+  },
+  resolve: async (
+    _: undefined,
+    { uuid }: IDeleteCompanyUser,
+    { currentUser }: IApolloServerContext
+  ) => {
+    const { companyUuid } = currentUser.getCompanyRole();
+    const companyUsers = await CompanyUserRepository.findByCompanyUuid(companyUuid);
+    if (companyUsers.length === 1) throw new DeleteLastCompanyUserError(companyUuid);
+
+    const companyUser = await CompanyUserRepository.findByUuid(uuid);
```

Comment:
porque no haces un find del array de users que ya tenes de la otra query?

Reply:
Porque si no existe en ese array no se si es porque quizo acceder a un companyUser de otra compay asi tiro error unauthorized o si es porque no existe que en ese caso tiro not found